### PR TITLE
Configurable chart colors

### LINKED ISSUE #352: Clarify order importantce for yAxisFields for charts
We should make it clear that `yAxisFields` order is important.

### LINKED ISSUE #353: Swap out primaryColor from default Chart colorset
The dataset that gets assigned primaryColor looks "special" because of how that color is used across the rest of the page. We should swap primaryColor out for something else.

### LINKED ISSUE #363: BarChart and LineChart should be importing from index
instead of from adjacent file

### PR DESCRIPTION
**Pull request by jondlm**
- `minor`: updated `BarChart` and `LineChart` to support alternate color palettes and custom colors
  - added `palette` prop
  - added `colorMap` prop
  - changed `xAxisTitleColor`, `yAxisTitleColor`, and `y2AxisTitleColor` to take custom colors, this was a _non-breaking_ change since the component still accepts a `number` for backward compatability
  - exposing a new set of color and palette constants on `lucid.chartConstants` for use with charts

Fixes #353 - fixes #369 - fixes #352 - fixes #363 

## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [ ] One core team UX approval

### CODE REVIEW

File: gulp/docs.js [hunk 1]
```diff
@@ -171,6 +172,10 @@ module.exports = {
 					}
 
 					var currentBranch = stdoutBranch.trim().replace(/\//g, '-'); // clean branch and replace forward slashes
+					var isTagged = !err && currentBranch === 'master';
```
Comment (jondlm) [inline gulp/docs.js]: This should fix the issue you were having @sodiumjoe when you were doing docs upload from a branch without a commit.

File: gulp/docs.js [hunk 2]
```diff
@@ -171,6 +172,10 @@ module.exports = {
...
+					var buildId = isTagged ? tag : currentBranch;
+
+					console.log('Uploading to docspot as ' + buildId);
```
Comment (jondlm) [inline gulp/docs.js]: Also logging what the doc will be named in docspot

File: src/components/Lines/examples/custom-colors.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	render() {
+		return (
+			<div>
+				<svg width={width} height={height}>
+					<Lines
+						data={data}
+						xScale={xScale}
+						yScale={yScale}
+						yFields={yFields}
+						colorMap={{
```
Comment (sodiumjoe) [inline src/components/Lines/examples/custom-colors.jsx]: 👍

File: src/components/AxisLabel/AxisLabel.jsx [hunk 4]
```diff
@@ -61,16 +72,25 @@ const AxisLabel = createClass({
...
+		const isCustomColor = _.startsWith(color, '#');
+		const colorStyle = isCustomColor ? { fill: color } : null;
 
 		return (
 			<text
 				{...passThroughs}
-				className={cx(className, '&', `&-color-${color % 6}`)}
+				style={{
+					...style,
+					...colorStyle,
```
Comment (ogupte) [inline src/components/AxisLabel/AxisLabel.jsx]: i think the order here should be swapped...if the consumer wants to overwrite the colorStyle with their own `style`, they should be able to.

File: src/components/Bar/examples/regular-colors.jsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	chartConstants.COLOR_BAD_DARK,
+	chartConstants.COLOR_NEUTRAL,
+];
+
+export default React.createClass({
+	render() {
+		return (
+			<div>
+				{_.map(colors, (color, i) =>
+					<svg key={i} {...svgProps}>
```
Comment (ogupte) [inline src/components/Bar/examples/regular-colors.jsx]: can't you just use `color` as the key here?

Comment (sodiumjoe) [inline src/components/Bar/examples/regular-colors.jsx]: haha, i was going to say the same thing, but it's just an example

File: src/components/AxisLabel/AxisLabel.jsx (hunk 4 above)
Comment (jondlm) [inline src/components/AxisLabel/AxisLabel.jsx]: Good catch!

**Pull request merged**